lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1304,6 +1319,43 @@ silOptModeArgStr(OptimizationMode mode) {
   }
 }
 
+static std::unique_ptr<UnifiedStatsReporter>
+computeStatsReporter(const CompilerInvocation &Invocation, CompilerInstance *Instance) {
```

Comment:
I'll let @graydon say if this part makes sense.

Reply:
Sorry, that should be "I'll let @graydon review this part". I assume you wouldn't write it to not make sense. :-)

Reply:
No need to apologize. I understood and agreed.

Reply:
Just code movement right? Factoring out the construction? Looks fine to me (and there are a few tests in the testsuite that will fail if there's an error in this refactoring that causes the stats reporter to stop working).

Reply:
Yes, intended to be just code movement. Thanks.